t/17-basetest.t

```diff
@@ -220,6 +221,60 @@ subtest record_screenmatch => sub {
             }
     ], 'screenmatch detail recorded as expected')
       or diag explain $basetest->{details};
+
+    # check a needle has workaround property
+    my $basetest_for_workaround = basetest->new();
+    my $misc_needle_dir         = dirname(__FILE__) . '/misc_needles/';
+    my $needle_file             = $misc_needle_dir . "check-workround-hash-bsc7654321-20190522.json";
+    my %workaround_match        = (
+        area => [
+            {x => 1, y => 2, w => 3, h => 4, result => 'ok'},
+        ],
+        error  => 0.25,
+        needle => {
+            name       => 'check-workround-hash-bsc7654321-20190522',
+            file       => $needle_file,
+            properties => [
+                {
+                    name  => 'workaround',
+                    value => 'bsc#7654321: this is a test about workaround.',
+                }
+            ],
+        },
+    );
```

Comment:
I suggest to just check what we care about here, i.e. not `is_deeply` but rather just check for the right keys and values being present

Reply:
Except for area and error, almost others need to be checked. so I prefer to check all hash.